File: leetcode_java/330. Patching Array.java
class Solution {
    public int minPatches(int[] nums, int n) {
        int i = 0, count = 0;
        long add = 1;
        while(add <= n){
            if(i < nums.length && nums[i] <= add) add += nums[i++];
            else {
                add += add;
                count ++;
            }
        }
        return count;
    }
}

File: leetcode_python/330. Patching Array.py
class Solution:
    def minPatches(self, nums: [int], n: int) -> int:
        add, i, count = 1, 0, 0
        while add <= n:
            if i < len(nums) and nums[i] <= add:
                add += nums[i]
                i += 1
            else:
                add += add
                count += 1
        return count


s = Solution()
s.minPatches([1,2,3,8], 80)